src/panels/config/tags/dialog-tag-detail.ts

```diff
@@ -191,13 +234,43 @@ class DialogTagDetail extends LitElement implements HassDialog {
     }
   }
 
+  private async _generateQR() {
+    const qrcode = await import('qrcode');
+    const canvas = await qrcode.toCanvas(
+      `https://home-assistant.io/tag/${this._params?.entry?.id}`,
+      {
+        width: 180,
+        errorCorrectionLevel: "Q",
+      }
+    );
+    const context = canvas.getContext("2d");
+
+    const imageObj = new Image();
+    imageObj.src = QR_LOGO_URL;
+    await new Promise((resolve) => {
+      imageObj.onload = resolve;
+    });
+    context.drawImage(
+      imageObj,
+      canvas.width / 3,
+      canvas.height / 3,
+      canvas.width / 3,
+      canvas.height / 3
+    );
+
+    this._qrCode = html`<img src=${canvas.toDataURL()}></img>`;
```

Comment:
We should set `_qrCode` to undefined when the dialog is closed.

Reply:
It's set to undefined when the dialog is opened. Both? Or only on close?

Reply:
On close then, as it saves memory.